### ISSUE TITLE
MacroCompiler-generated mem IOs don't match ReplSeqMem-generated

### ISSUE THREAD

**Issue by jamesdunn** [ucb-bar/barstools#40]
ReplSeqMem and MacroCompiler generate different IOs for memory modules. For example, the blackbox wrapper in DigitalTop.v for a BOOM build contains the following IOs:

```
data_arrays_0_ext data_arrays_0_ext (          
  .RW0_addr(data_arrays_0_ext_RW0_addr),       
  .RW0_en(data_arrays_0_ext_RW0_en),           
  .RW0_clk(data_arrays_0_ext_RW0_clk),         
  .RW0_wmode(data_arrays_0_ext_RW0_wmode),     
  .RW0_wdata(data_arrays_0_ext_RW0_wdata),     
  .RW0_rdata(data_arrays_0_ext_RW0_rdata),     
  .RW0_wmask(data_arrays_0_ext_RW0_wmask)    
);
```

The mems.v generated by MacroCompiler for `st28fdsoi-lib.json` contains a mapped memory  that looks like the following:

```
module data_arrays_0_ext(     
input         RW0_clk,      
input  [7:0]  RW0_addr,     
input  [31:0] RW0_wdata,    
output [31:0] RW0_rdata,    
input         RW0_ren,      
input         RW0_wen,      
input  [3:0]  RW0_mask    
);
```

The latter contains `RW0_ren` and `RW0_wen`, while the former has IO for just `RW0_en`, among other IO discrepancies.

**Comment by jamesdunn** [comment]
@jwright6323 @edwardcwang @colinschmidt 
I think I may have found the source of this.

For an ST28 digital top, using the previous `hammer-generate-mdf` created a `mem.json` from `mem.conf` that has these IOs:
```
 "depth": 256,
 "name": "data_arrays_0_ext",
 "ports": [
     {
         "address port name": "RW0_addr",
         "chip enable port name": "RW0_en",
         "clock port name": "RW0_clk",
         "input port name": "RW0_wdata",
         "mask granularity": 8,
         "mask port name": "RW0_wmask",
         "output port name": "RW0_rdata",
         "write enable port name": "RW0_wmode"
    }
```
Though for the same `DigitalTop.conf` with the new MacroCompiler that eliminates `hammer-generate-mdf`, I get a final `mems.v` with the following IOs:
```
module data_arrays_0_ext(     
input         RW0_clk,      
input  [7:0]  RW0_addr,     
input  [31:0] RW0_wdata,    
output [31:0] RW0_rdata,    
input         RW0_ren,      
input         RW0_wen,      
input  [3:0]  RW0_mask    
);
```
There is an `RW0_ren` and `RW0_wen` instead of a `RW0_en`.

Part of #35 included a modification to MacroCompiler to ingest .confs directly. Looking at [this change](https://github.com/ucb-bar/barstools/commit/12842cb3a79bb6d59d61ebb2af17c4a67fd42d6e), I'm wondering if the .conf ingestion made assumptions about the IO ports of the vendor SRAMs. Do these IOs differ from vendor to vendor?

**Comment by edwardcwang** [comment]
@jamesdunn I think your analysis is spot-on here. The issue is that the barstools conf-to-mdf converter is buggy in that it assumes separate `ren` and `wen` ports. If you look in the generate-mdf command you will see no such thing: https://github.com/ucb-bar/hammer/blob/master/src/hammer-shell/hammer-generate-mdf

**Comment by jwright6323** [comment]
@jamesdunn I think you're off-master somehow. This was fixed before #35 was merged: https://github.com/ucb-bar/barstools/pull/35/commits/1789a49e8d30fc563c32a0eda83151d35028e84b

@edwardcwang It's fine here too: https://github.com/ucb-bar/barstools/blob/9d505d6063f07f7750686f67d2cda49b17f6d898/macros/src/main/scala/Utils.scala#L134 You should check the code before you post such a passive-aggressive comment.

**Comment by edwardcwang** [comment]
Sorry about that. I was working off the commit James provided and didn't realize that he was off master. My bad.
